src/libraries/System.Private.CoreLib/src/System/Numerics/TotalOrderIeee754Comparer.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Numerics
+{
+    /// <summary>
+    /// Represents a comparison operation that compares floating-point numbers
+    /// with IEEE 754 totalOrder semantic.
+    /// </summary>
+    /// <typeparam name="T">The type of the numbers to be compared, must be an IEEE 754 floating-point type.</typeparam>
+    public struct TotalOrderIeee754Comparer<T> : IComparer<T>, IEqualityComparer<T>
```

Comment:
The struct has no state and should never have mutable state:
```suggestion
    public readonly struct TotalOrderIeee754Comparer<T> : IComparer<T>, IEqualityComparer<T>
```

This will ensure that it doesn't create any copies for various patterns a user might use/consider.